docs/fxa-engineering/fxa-webchannel-protocol.md

```diff
@@ -0,0 +1,324 @@
+---
+id: fxa-webchannel-protocol
+title: WebChannel Protocol
+sidebar_label: WebChannel Protocol
+---
+
+Current as of `December 2nd, 2019`
+
+# Communication with Firefox via WebChannels
+
+Starting with Firefox 40, Firefox Accounts communicates with Firefox desktop and Fennec using [WebChannels](https://developer.mozilla.org/en-US/docs/Mozilla/JavaScript_code_modules/WebChannel.jsm). WebChannels provide secure two-way communication between the browser and web content. Messages sent from content to the browser utilize `WebChannelMessageToChrome` custom events. Messages sent from the browser to content utilize `WebChannelMessageToContent` custom events.
+
+For WebChannel usage in the Fenix browser, see [FxA/OAuth WebChannel Protocol](./fxa-oauth-webchannel-protocol).
+
+### Context
+
+WebChannels are used to communicate Sync related information when
+`service=sync&context=<context>` query parameters are specified.
+`<context>` should be one of:
+
+- `fx_desktop_v3`
+- `fx_fennec_v1`
+
+### Commands
+
+A command is a `WebChannelMessageToChrome` custom event sent from FxA to the browser using the `account_updates` WebChannel.
+
+Command format is explained in the section titled [Command/Request format](#command-request-format).
+
+#### fxaccounts:fxa_status
+
+Sent on startup to fetch the FxA state from supporting browsers.
+Used to get the browser's currently signed in user as well as browser capabilities.
+
+##### request data
+
+```js
+{
+  isPairing: <boolean>
+  service: <string>
+}
+```
+
+- `isPairing`: whether or not accounts.firefox.com is executing a pairing flow
+- `service`: the service name or OAuth client-id that is requesting authentication
+
+##### response data
+
+```js
+{
+  capabilities: <capabilities>
+  signedInUser: <signed in user> || null
+}
+```
+
+- `capabilities` - A list of capabilities supported by the browser. See [capabilities](#capabilities).
+- `signedInUser` - the user currently signed into Sync on the browser. See [signedInUser](#signedInUser).
+
+##### support
+
+Fx Desktop 55+
+
+#### fxaccounts:loaded
+
+Sent on startup. Sent after the first screen is visible. No data is sent. No response is expected.
+
+#### fxaccounts:can_link_account
+
+Sent on signin or signup after a user has entered their credentials and submit the form. A response with an `ok` field is expected.
+
+##### request data
+
+```js
+{
+  email: <email>
+}
+```
+
+##### response data
+
+```js
+{
+  ok: true | false;
+}
+```
+
+#### fxaccounts:login
+
+Sent when a user successfully authenticates with Firefox Accounts and sync can begin. This includes after a user signs up but before they verify their email address, after an account unlock, and after a password reset. No response is expected.
+
+##### data
+
+See [Login Data](#logindata).
+
+#### fxaccounts:oauth_login
+
+Sent when a user successfully authenticates via a WebChannel OAuth flow.
```

Comment:
This one feels like it should be moved into the oauth-specific doc.

Reply:
Moved to the OAuth doc.